src/trading/NewOrdersFetcher.sol

```diff
@@ -0,0 +1,148 @@
+/**
+ * Copyright (C) 2015 Forecast Foundation OU, full GPL notice in LICENSE
+ */
+
+pragma solidity ^0.4.13;
+
+import 'ROOT/Controller.sol';
+import 'ROOT/libraries/arrays/Bytes20Arrays.sol';
+import 'ROOT/libraries/Trading.sol';
+import 'ROOT/reporting/Interfaces.sol';
+import 'ROOT/trading/NewOrders.sol';
+
+
+/**
+ * @title OrdersFetcher
+ */
+// FIXME: This was named NewOrdersFetcher to prevent naming conflicts with ordersFetcher.se. Should be renamed to OrdersFetcher.
+contract NewOrdersFetcher is Controlled {
+    using Bytes20Arrays for bytes20[];
+
+    // FIXME: Replace these constants with Trading.TradeTypes enum once all contracts are migrated to Solidity.  (This was done because Serpent contracts could not pass in uint8 variables as rading.TradeTypes parameters.)
+    uint8 private constant BID = 1;
+    uint8 private constant ASK = 2;
+
+    /**
+     * @dev Get orders for a particular market, type, and outcome (chunked)
+     */
+    function getOrderIds(uint8 _type, Market _market, uint8 _outcome, bytes20 _startingOrderId, uint256 _numOrdersToLoad) public constant returns (bytes20[]) {
+        require(_type == BID || _type == ASK);
+        require(0 <= _outcome && _outcome < _market.getNumberOfOutcomes());
+        NewOrders _orders = NewOrders(controller.lookup("NewOrders"));
+        if (_startingOrderId == 0) {
+            _startingOrderId = _orders.getBestOrderId(_type, _market, _outcome);
+        }
+        bytes20[] _orderIds;
+        _orderIds[0] = _startingOrderId;
+        uint256 _i = 0;
+        while (_i < _numOrdersToLoad && _orders.getWorseOrderId(_orderIds[_i], _type, _market, _outcome) != 0) {
+            _orderIds[_i + 1] = _orders.getWorseOrderId(_orderIds[_i], _type, _market, _outcome);
+            _i += 1;
+        }
+        return (_orderIds.slice(0, _i));
+    }
+
+    function getOrder(bytes20 _orderId, uint8 _type, Market _market, uint8 _outcome) public constant returns (uint256 _attoshares, int256 _displayPrice, address _owner, int256 _tokensEscrowed, uint256 _sharesEscrowed, bytes20 _betterOrderId, bytes20 _worseOrderId, uint256 _gasPrice) {
```

Comment:
`tokensEscrowed` should be `uint256`.

Reply:
It's being set to orders.getOrderMoneyEscrowed(), which returns fxpMoneyEscrowed.  I thought you had said previously that this should be int256: https://augurproject.slack.com/archives/C031NRD9B/p1503637460000088.  Is this not the case?

Reply:
Price should be `int256` (for the time being), but the amounts being escrowed can never be negative.